### PR TITLE
Expose getPageProps as public client method

### PR DESCRIPTION
**Pull request by gryphonmyers**

### CODE REVIEW

**Comment by brillout** [comment]
Neat. Do not hesitate to hit me with questions :-).

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,6 +5,9 @@
 - [Run Test Suite](#run-test-suite)
 
 
+## Environment Requirements
+- Node.js 15.*.*
```
Comment (chrisvariety) [inline CONTRIBUTING.md:9]: This is fairly unrealistic, node v15 will never be LTS. Could wait for node v16, which is slated for ~4/20.

Comment (gryphonmyers) [reply CONTRIBUTING.md:9]: Should only be a development requirement, not a deployment requirement

Comment (chrisvariety) [reply CONTRIBUTING.md:9]: ah, my bad, missed it was in that file, 👍

**Comment by gryphonmyers** [comment]
@brillout I changed the name of this PR to describe a more atomic feature. This is a feature that is working the way I want it to now and I think largely isolated from any other changes I'll be proposing, so I'm marking it as ready for review.

**Comment by brillout** [comment]
LGTM. Do you need this published or are you fine with `yarn link` / `npm link`?

**Comment by gryphonmyers** [comment]
A publish would be nice actually. Quick q on tests though - should we have unit tests covering this functionality? This feature is definitely something I could cover with a unit test. It seems like the tests in the project right now are more like end to end tests (I didn't fully get them running, but it seemed like it was installing a headless browser(?)) Could you explain a bit about how you'd like to approach testing with this project?

**Comment by brillout** [comment]
GitHub doesn't say so but this PR is actually merged.

There was a bug in the PR so I made some modifications. AFAICT my modifications should still achieve what you need and, while I was at it, I also implemented additional low-level interfaces:
 - `import { getPageById } from 'vite-plugin-ssr/client'`
 - `import { getPageId } from 'vite-plugin-ssr/client/router'`
 - `import { getPageProps } from 'vite-plugin-ssr/client/router'`
 - `import { getPageByUrl } from 'vite-plugin-ssr/client/router'`

Note that, contrary to your PR, `getPageProps(url: string)` is available at `vite-plugin-ssr/client/router`.

Let me know if these new functions work for you.

Yes for now there are only end-to-end tests but I've got some plans to add more granular tests in the near future.

Let's add a test for these new interfaces once we have a working implementation of a deep Vue Router integration that is documented in the `README.md`.

> I didn't fully get them running

I improved the `CONTRIBUTING.md` and npm scripts.

**Comment by brillout** [comment]
Released in `v0.1.0-beta.27`.

**Pull request closed without merging**